spec.html

```diff
@@ -936,6 +989,24 @@ <h1>The AsyncContext Object</h1>
 
     <emu-clause id="sec-asynccontext-abstract-operations">
       <h1>AsyncContext Abstract Operations</h1>
+
+      <emu-clause id="sec-gethostdefinedasynccontextmapping" type="abstract operation">
+        <h1>GetHostDefinedAsyncContextMapping ( ): a List of Async Context Mapping Records</h1>
+        <dl class="header">
+        </dl>
+        <p>An implementation of GetHostDefinedAsyncContextMapping must conform to the following requirements:</p>
+        <ul>
+          <li>It must perform and return a List of Async Context Mapping Records.</li>
```

Comment:
Another requirement would be to always return the same mapping.

Reply:
or in the opposite direction, pass the module record to it.

Reply:
Updated with a parameter module records and the result list must contain the same mapping for the given module record.

Reply:
I'm not sure I understand this change. I thought the goal of this PR was to make is so that all modules have the same top-level mapping, while this PR is now defining the top-level mapping as potentially different for each module.

Reply:
This allows hosts to add mapping entries based on the origin of the module records. The goal of the PR is allowing hosts initialize their entries.

Reply:
So this allows hosts to implement both fresh-context-per-module and module-based-on-importer?

Reply:
Not sure what you mean by module-based-on-importer, but my understanding is that if two different importers import the same module, you'd still only evaluate once, and any importer dependency would resolve based on one or the other (presumably the earlier one).

Reply:
Right, the original motivation of this PR was to give an answer to this question: what is the context of the module in this case?

```js
variable.get(); // undefined

variable.run(1, () => import("x"));
variable.run(2, () => import("x"));
```

At the top-level of `x`, does `variable.get()` return `undefined`, `0`, or `2`? This PR as-is does not give an answer, and instead laves it host-defined.